autotech.lua

```diff
@@ -390,20 +391,79 @@ function auto_tech:set_tech_unit()
         error()
     end
 
+    --get the depths of all science packs on the tech tree, assuming non-progression military science
+    self.technology_nodes:for_all_nodes(function(technology_node)
+        local factorio_tech = technology_node.object_node.object
+        local science_pack_unlocked_by_this_tech = data.raw.tool[technology_node.object_node.object.name]
+        if science_pack_unlocked_by_this_tech then
+            if factorio_tech.name == "military-science-pack" then
+                if verbose_logging then
+                    log("Depth of a military science pack tech is " .. technology_node.depth)
+                end
+                goto continue
+            end
+            depths_of_science_packs[#depths_of_science_packs+1]=technology_node.depth
+            if verbose_logging then
+                log("Depth of a new science pack tech is " .. technology_node.depth)
+            end
+            ::continue::
+        end
+    end)
+
+    --sort the table of depths from lowest to highest
+    table.sort(depths_of_science_packs)
+    for i,pack_depth in pairs(depths_of_science_packs) do
+        if verbose_logging then
```

Comment:
Instead of checking every iteration, run the loop only if `verbose_logging` is `true`